File: common/mrtjp/projectred/integration2/GateItemRenderer.java
package mrtjp.projectred.integration2;

import net.minecraft.item.ItemStack;
import net.minecraftforge.client.IItemRenderer;
import codechicken.lib.render.CCRenderState;
import codechicken.lib.render.TextureUtils;
import codechicken.lib.vec.Scale;
import codechicken.lib.vec.TransformationList;
import codechicken.lib.vec.Translation;

public class GateItemRenderer implements IItemRenderer
{
    public final static GateItemRenderer instance = new GateItemRenderer();
    
    @Override
    public boolean handleRenderType(ItemStack item, ItemRenderType type) {
        return true;
    }

    @Override
    public boolean shouldUseRenderHelper(ItemRenderType type, ItemStack item, ItemRendererHelper helper) {
        return true;
    }

    @Override
    public void renderItem(ItemRenderType type, ItemStack item, Object... data) {
        int damage = item.getItemDamage();
        switch (type) {
        case ENTITY:
            renderGateInv(damage, -.3f, 0f, -.3f, .6f);
            return;
        case EQUIPPED:
            renderGateInv(damage, 0f, .15f, 0f, 1f);
            return;
        case EQUIPPED_FIRST_PERSON:
            renderGateInv(damage, 1f, -.2f, -.4f, 2f);
            return;
        case INVENTORY:
            renderGateInv(damage, 0f, .20f, 0f, 1f);
            return;
        default:
            return;
        }
    }

    public void renderGateInv(int meta, float x, float y, float z, float scale) {
        TextureUtils.bindAtlas(0);
        CCRenderState.reset();
        CCRenderState.useNormals(true);
        CCRenderState.pullLightmap();
        CCRenderState.setColourOpaque(-1);
        TransformationList t = new TransformationList();
        t.with(new Scale(scale)).with(new Translation(x, y, z));
        RenderGate.renderInv(t, meta);
    }
}

File: src/main/scala/mrtjp/projectred/integration/GateItemRenderer.scala
package mrtjp.projectred.integration

import codechicken.lib.render.CCRenderState
import codechicken.lib.render.item.IItemRenderer
import codechicken.lib.util.TransformUtils
import codechicken.lib.vec.RedundantTransformation
import com.google.gson.{JsonDeserializationContext, JsonObject}
import com.mojang.blaze3d.matrix.MatrixStack
import com.mojang.datafixers.util.Pair
import net.minecraft.client.renderer.model._
import net.minecraft.client.renderer.texture.TextureAtlasSprite
import net.minecraft.client.renderer.{IRenderTypeBuffer, RenderType}
import net.minecraft.item.ItemStack
import net.minecraft.util.ResourceLocation
import net.minecraftforge.client.model.geometry.IModelGeometry
import net.minecraftforge.client.model.{IModelConfiguration, IModelLoader}

import java.util
import java.util.Collections
import java.util.function.{Function => JFunc}

object GateItemRenderer extends IItemRenderer
{
    override def isAmbientOcclusion = true

    override def isGui3d = true

    override def getTransforms = TransformUtils.DEFAULT_BLOCK

    override def func_230044_c_() = true

    override def renderItem(stack:ItemStack, transformType:ItemCameraTransforms.TransformType, mStack:MatrixStack, getter:IRenderTypeBuffer, packedLight:Int, packedOverlay:Int):Unit =
    {
        stack.getItem match {
            case gate:ItemPartGate =>
                val ccrs = CCRenderState.instance()
                ccrs.reset()
                ccrs.brightness = packedLight
                ccrs.overlay = packedOverlay
                ccrs.bind(RenderType.getCutout, getter, mStack)
                RenderGate.renderInv(stack, new RedundantTransformation,gate.gateType, ccrs)
        }
    }

    //    override def renderItem(item:ItemStack, transformType: TransformType)
    //    {
    //        val meta = item.getItemDamage
    //        if (!GateDefinition.values.isDefinedAt(meta) ||
    //                !GateDefinition(meta).implemented) return
    //
    //        enableBlend()
    //        blendFunc(GL11.GL_SRC_ALPHA, GL11.GL_ONE_MINUS_SRC_ALPHA)
    //
    //        TextureUtils.bindBlockTexture()
    //        val ccrs = CCRenderState.instance()
    //        ccrs.reset()
    //        ccrs.pullLightmap()
    //        RenderGate.renderInv(item, new RedundantTransformation, item.getItemDamage, ccrs)
    //
    //        disableBlend()
    //    }

    class Loader extends IModelLoader[Loader] with IModelGeometry[Loader] {
        override def read(deserializationContext: JsonDeserializationContext, modelContents: JsonObject) = this

        override def getTextures(owner: IModelConfiguration, modelGetter: JFunc[ResourceLocation, IUnbakedModel], missingTextureErrors: util.Set[Pair[String, String]]) = Collections.emptyList()

        override def bake(owner: IModelConfiguration, bakery: ModelBakery, spriteGetter: JFunc[Material, TextureAtlasSprite], modelTransform: IModelTransform, overrides: ItemOverrideList, modelLocation: ResourceLocation) = GateItemRenderer
    }
}
